src/hisp/plamsa_data_handling/main.py

```diff
@@ -20,69 +22,82 @@ def __init__(
         self._time_to_RISP_data = {}
 
     def get_particle_flux(
-        self, pulse_type: str, nb_mb: int, t_rel: float, ion=True
+        self, pulse_type: str, nb_bin: SubBin | DivBin, t_rel: float, ion=True
     ) -> float:
         """Returns the particle flux for a given pulse type
 
         Args:
             pulse_type: pulse type (eg. FP, ICWC, RISP, GDC, BAKE)
-            nb_mb: monoblock number
+            bin: SubBin or DivBin
             t_rel: t_rel as an integer (in seconds).
                 t_rel = t - t_pulse_start where t_pulse_start is the start of the pulse in seconds
             ion (bool, optional): _description_. Defaults to True.
 
         Returns:
             float: particle flux in part/m2/s
         """
+        if isinstance(nb_bin, SubBin):
+            bin_index = nb_bin.parent_bin_index
+            wetted_frac = nb_bin.wetted_frac
+        elif isinstance(nb_bin, DivBin):
+            bin_index = nb_bin.index
+            wetted_frac = 1
+
         if ion:
             FP_index = 2
             other_index = 0
+            flux_frac = wetted_frac
         if not ion:
             FP_index = 3
             other_index = 1
+            flux_frac = 1
```

Comment:
`wetted_frac` and `flux_frac` seem to be the same thing here no? I'm a bit confused. Maybe it's just missing clear documentation and comments